internal/e2e/e2e_index_test.go

```diff
@@ -558,149 +558,18 @@ func TestDeleteOldShards(t *testing.T) {
 	}
 }
 
-func TestPartialSuccess(t *testing.T) {
-	dir := t.TempDir()
-
-	opts := Options{
-		IndexDir:    dir,
-		ShardMax:    1024,
-		SizeMax:     1 << 20,
-		Parallelism: 1,
-	}
-	opts.RepositoryDescription.Name = "repo"
-	opts.SetDefaults()
-
-	b, err := NewBuilder(opts)
-	if err != nil {
-		t.Fatalf("NewBuilder: %v", err)
-	}
-
-	for i := 0; i < 4; i++ {
-		nm := fmt.Sprintf("F%d", i)
-		_ = b.AddFile(nm, []byte(strings.Repeat("01234567\n", 128)))
-	}
-	b.buildError = fmt.Errorf("any error")
-
-	// No error checking.
-	_ = b.Finish()
-
-	// Finish cleans up temporary files.
-	if fs, err := filepath.Glob(dir + "/*"); err != nil {
-		t.Errorf("glob(%s): %v", dir, err)
-	} else if len(fs) != 0 {
-		t.Errorf("got shards %v, want []", fs)
-	}
-}
-
-type filerankCase struct {
-	name string
-	docs []*zoekt.Document
-	want []int
-}
-
-func testFileRankAspect(t *testing.T, c filerankCase) {
-	var want []*zoekt.Document
-	for _, j := range c.want {
-		want = append(want, c.docs[j])
-	}
-
-	got := make([]*zoekt.Document, len(c.docs))
-	copy(got, c.docs)
-	sortDocuments(got)
```

Comment:
Also moved this test to `builder_test.go` to keep `sortDocuments` private, and since it's more of a unit test.